tools/lsp/ui/components/widgets/floating-color-picker-widget.slint

```diff
@@ -0,0 +1,426 @@
+                source: @image-url("../../assets/close.svg");
+            }
+        }
+
+        Rectangle {
+            width: 100%;
+            height: 1px;
+            x: 0;
+            y: parent.height - self.height;
+            background: Styles.divider-color;
+        }
+    }
+
+    saturation-value-holder := Rectangle {
+        y: title.y + title.height;
+        width: 100%;
+        height: self.width * 0.8;
+        saturation-value := Rectangle {
+            width: parent.width - 16px;
+            height: parent.height - 16px;
+
+            Rectangle {
+                border-radius: 3px;
+                border-width: 1px;
+                border-color: white.transparentize(90%);
+                clip: true;
+                hue := Rectangle {
+                    background: hsv(root.hue, 1, 1);
+                }
+
+                sat := Rectangle {
+                    background: @linear-gradient(90deg, white 0%, transparent 100%);
+                }
+
+                val := Rectangle {
+                    background: @linear-gradient(0deg, black 0%, transparent 100%);
+                }
+
+                TouchArea {
+                    moved => {
+                        if self.has-hover {
+                            root.saturation = clamp(self.mouse-x / self.width, 0, 1);
+                            root.value = clamp(1 - (self.mouse-y / self.height), 0, 1);
+                        }
+                    }
+                    changed pressed => {
+                        if self.pressed {
+                            root.saturation = self.mouse-x / self.width;
+                            root.value = 1 - (self.mouse-y / self.height);
+                        }
+                    }
+                }
+            }
+
+            sv-picker := Rectangle {
+                x: 1px + (saturation-value.width - 3px) * root.saturation;
+                y: 1px + (saturation-value.height - 3px) * (1 - root.value);
+                width: 1px;
+                height: 1px;
+                Rectangle {
+                    width: 12px;
+                    height: 12px;
+                    border-radius: self.width / 2;
+                    background: white;
+                    drop-shadow-blur: 5px;
+                    drop-shadow-offset-y: 2px;
+                    drop-shadow-color: #000000b4;
+                }
+
+                Rectangle {
+                    width: 8px;
+                    height: 8px;
+                    border-radius: self.width / 2;
+                    background: root.current-color;
+                }
+            }
+        }
+    }
+
+    hsva-controls := Rectangle {
+        y: saturation-value-holder.y + saturation-value-holder.height;
+        width: 100%;
+        height: 56px;
+        // The following properties are used to size the hue picker and control the
+        // thumb to now go past the visual bounds. But the TouchArea is intentionally larger
```

Comment:
Is this supposed to say `now go past` or was it meant to say `not go past`? Just a nitpick.